### PR TITLE
dotnet EOL warning suppression

### PR DESCRIPTION
**Pull request by CharlieSaunders**
Adding a property to the dotnet run process that will suppress EOL Framework warnings based on the bool from the config.

When using the package on older frameworks, the dotnet run process; rightly, displays that the framework being used has reached EOL. This however means that the JSON generated is invalid as it contains warnings IE:

C:\dotnet\sdk\7.0.202\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.EolTargetFrameworks.targets(28,5): warning NETSDK1138: The target framework 'netcoreapp2.0' is out of support and will not receive security updates in the future. Please refer to https://aka.ms/dotnet-core-support for more information about the support policy. [C:\path\to\the\node_modules\csharp-models-to-typescript\lib\csharp-models-to-json\csharp-models-to-json.csproj]
[{foo:bar}]

Which when ran through JSON.parse; index.js: 70, errors with "Unexpected token C in JSON at position 0". 

Adding this additional parameter on the config; suppressEolFrameworkErrors, suppresses those EOL warnings and only the generated JSON is sent for parsing.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -45,7 +45,13 @@ const converter = createConverter({
...
 
 const dotnetProject = path.join(__dirname, 'lib/csharp-models-to-json');
-const dotnetProcess = spawn('dotnet', ['run', `--project "${dotnetProject}"`, `"${path.resolve(configPath)}"`], { shell: true });
+
+let processParamaters = ['run', `--project "${dotnetProject}"`, `"${path.resolve(configPath)}"`];
+if(config.suppressEolFrameworkErrors){
+    processParamaters.push('--property:CheckEolTargetFramework=false');
+}
+
+const dotnetProcess = spawn('dotnet', processParameters, { shell: true });
```
Comment (digocesar) [inline index.js:54]: Hi @CharlieSaunders .
The variable `processParameters` is erronealy typed before as processParam**a**ters. Can you fix it?

File: index.js [hunk 2]
```diff
@@ -45,7 +45,13 @@ const converter = createConverter({
 let timer = process.hrtime();
 
 const dotnetProject = path.join(__dirname, 'lib/csharp-models-to-json');
-const dotnetProcess = spawn('dotnet', ['run', `--project "${dotnetProject}"`, `"${path.resolve(configPath)}"`], { shell: true });
+
+let processParamaters = ['run', `--project "${dotnetProject}"`, `"${path.resolve(configPath)}"`];
+if(config.suppressEolFrameworkErrors){
```
Comment (digocesar) [inline index.js:50]: May be put space caracter after the `if` and before `{` to be bether formmated. 😉

**Review by digocesar — CHANGES_REQUESTED** [review]
Please rebase this branch to the current `master` that has automated tests.

**Comment by digocesar** [comment]
@svenheden and/or @SafeerH.
If dotnet process has this option to ignore EOL error, is not better to always pass this parameter to do not emmit the EOL error?
We had updated `main` to .Net 8.0, but this parameter can avoid the component stops working in the future, as happened recently.

**Comment by svenheden** [comment]
> If dotnet process has this option to ignore EOL error, is not better to always pass this parameter to do not emmit the EOL error?

Yes that sounds like a good idea. I can't think of a scenario where you'd want to show that EOL error causing the JSON to be corrupted.

**Comment by SafeerH** [comment]
The root cause for the problem is the way of extracting the JSON from the console output.
If there's anything else apart from the JSON string in the Console output (stdout), the program breaks. Also relates to #61 
So, I think we need to find a broader solution that will isolate the JSON irrespective of other console outputs. 
I'm thinking of printing the JSON between markers, that will help us isolate the JSON.
I will try this approach and let you guys know. Thanks!

**Review by digocesar — APPROVED** [review]

**Comment by digocesar** [comment]
Hi @CharlieSaunders. Thanks for your help.
This PR is not needed anymore as the #78 solved the problem.

**Pull request closed without merging**